components/reusable-components/navigation/Navigation.tsx

```diff
@@ -3,49 +3,125 @@
 
 'use client';
 
-import React from 'react';
+import React, { useState, useRef } from 'react';
 import Link from 'next/link';
 import Image from 'next/image';
-import 'bootstrap-icons/font/bootstrap-icons.css';
-
-import styles from './navigation.module.scss';
-import { useTheme } from '../../../app/context/themeContext';
+import { usePathname } from 'next/navigation';
+import { FaUser, FaSignOutAlt, FaChevronDown, FaChevronUp, FaEnvelope } from 'react-icons/fa';
 import sunImage from '../../../public/theme/sun.png';
 import moonImage from '../../../public/theme/moon.png';
 import LogoNavigation from '../../../public/logo/logo-black.svg';
+import styles from './navigation.module.scss';
+import { useTheme } from '../../../app/context/themeContext';
+import { User } from '../_Types';
+import useClickOutside from '../../../api/utils/HOC/useClickOutside';
+import LogoutButton from '../button/LogoutButton';
+
+const user: User = {
+  email: 'john@example.com',
+  first_name: 'John',
+  last_name: 'Doe',
+  status: 'active',
+  image: null,
+};
```

Comment:
user should not be hardcoded, we can get the the logged in user.

Reply:
Login functionality wasn't implemented when this was built, I will change it now